numpyro/mcmc.py

```diff
@@ -405,11 +401,14 @@ def mcmc(num_warmup, num_samples, init_params, num_chains=1, sampler='hmc',
         potential_fn = sampler_kwargs.pop('potential_fn')
         kinetic_fn = sampler_kwargs.pop('kinetic_fn', None)
         algo = sampler_kwargs.pop('algo', 'NUTS')
-        rngs = sampler_kwargs.pop('rng', vmap(PRNGKey)(np.arange(num_chains)))
+        if num_chains > 1:
+            rngs = sampler_kwargs.pop('rng', vmap(PRNGKey)(np.arange(num_chains)))
+        else:
+            rng = sampler_kwargs.pop('rng', PRNGKey(0))
 
         init_kernel, sample_kernel = hmc(potential_fn, kinetic_fn, algo)
         if progbar:
-            hmc_state = init_kernel(init_params, num_warmup, progbar=progbar, rng=rngs[0],
+            hmc_state = init_kernel(init_params, num_warmup, progbar=progbar, rng=rng,
                                     **sampler_kwargs)
             samples_flat = fori_collect(0, num_samples, sample_kernel, hmc_state,
```

Comment:
Not related to this PR, but could you add `progbar_desc='sample'` in `single_chain_mcmc`?

Reply:
I just check: that `single_chain_mcmc` does not use `progbar` so I think that adding `progbar_desc='sample'` does not take effect.

Reply:
Sorry, my bad. I saw this in some test, but that uses fori_collect directly, so it isn't related.